### ISSUE TITLE
Exclude directories from Dialyzer checking

### ISSUE THREAD

**Issue by onno-vos-dev** [erlang-ls/erlang_ls#468]
**Is your feature request related to a problem? Please describe.**
Dialyzer as we all know is somewhat time consuming. Especially for large(r) files or files with lots of references, running dialyzer using source code only, can be somewhat time consuming. For certain files, I may however not be interested in seeing the dialyzer results in my editor.

**Describe the solution you'd like**
I'd like to be able to pass a list of folders (with wildcards) in the `erlang_ls.config` that would allow skipping dialyzer for certain directories. 

**Describe alternatives you've considered**
N/A

(Happy to work on this feature but would appreciate a :+1: or :-1: )

**Comment by jfacorro** [comment]
Have you noticed that the presentation of diagnostics gets delayed specifically because of dialyzer? If so how big is the file you were editing (maybe in lines of code)? I ask so other people can reproduce the behavior.

**Comment by onno-vos-dev** [comment]
Exactly, both the presentation gets delayed and also the code_reload gets delayed. I noticed in a common test suite of 1000 lines which touches the core the application I work with so Dialyzer typically tends to spider out. 

The code_reload getting delayed is what caught my attention since after making a change, I re-ran my tests yet they still failed. 2nd time I ran, the code was reloaded and the tests passed. Since the code_reload happens after the initial compile, dialyzer and elvis checks, those parts delay the code_reload.

I'm not sure if the following would work but otherwise, this might be worth considering:
- `els_compiler_diagnostics:diagnostics/1` is the only dependency for `els_text_synchronization:maybe_compile_and_load/2`
- Both `els_dialyzer_diagnostics:diagnostics/1` and `els_elvis_diagnostics:diagnostics/1` instead both get spawned and send their diagnostics back separately. Both of these, despite being interesting for the developer, IMHO are perfectly fine to come with a minor delay. I'm however not sure what happens to these messages when they're handling by the language server. Do they overwrite eachother?
